src/lib/homestore_backend/hs_blob_manager.cpp

```diff
@@ -1,17 +1,271 @@
+    RELEASE_ASSERT(index_table != nullptr, "Index table instance null");
+
+    auto r = get_from_index_table(index_table, shard.id, blob_id);
+    if (!r) {
+        LOGWARN("Blob not found in index id {} shard {}", blob_id, shard.id);
+        return folly::makeUnexpected(r.error());
+    }
+
+    auto multi_blkids = r.value();
+    auto block_size = repl_dev->get_blk_size();
+    auto sgs_ptr = std::make_shared< sisl::sg_list >();
+    auto total_size = multi_blkids.blk_count() * block_size;
+    auto iov_base = iomanager.iobuf_alloc(block_size, total_size);
+    sgs_ptr->size = total_size;
+    sgs_ptr->iovs.emplace_back(iovec{.iov_base = iov_base, .iov_len = total_size});
+
+    return repl_dev->async_read(multi_blkids, *sgs_ptr, total_size)
+        .thenValue([block_size, blob_id, req_len, req_offset, shard, multi_blkids, sgs_ptr,
+                    iov_base](auto&& result) mutable -> BlobManager::AsyncResult< Blob > {
+            if (result) {
+                LOGERROR("Failed to read blob {} shard {} err {}", blob_id, shard.id, result.value());
+                iomanager.iobuf_free(iov_base);
+                return folly::makeUnexpected(BlobError::READ_FAILED);
+            }
+
+            BlobHeader* header = (BlobHeader*)iov_base;
+            if (!header->valid() || header->shard_id != shard.id) {
+                LOGERROR("Invalid header found for blob {} shard {}", blob_id, shard.id);
+                LOGERROR("Blob header {}", header->to_string());
+                iomanager.iobuf_free(iov_base);
+                return folly::makeUnexpected(BlobError::READ_FAILED);
+            }
+
+            auto header_size = sisl::round_up(sizeof(BlobHeader), block_size);
+            size_t blob_size = header->meta_data_offset == 0 ? header->total_size : header->meta_data_offset;
+            uint8_t* blob_bytes = (uint8_t*)iov_base + header_size;
+            if (!header->compute_blob_hash(sisl::blob{blob_bytes, (uint32_t)blob_size}, true /* verify */)) {
+                LOGERROR("Hash mismatch for blob {} shard {}", blob_id, shard.id);
+                iomanager.iobuf_free(iov_base);
+                return folly::makeUnexpected(BlobError::CHECKSUM_MISMATCH);
+            }
+
+            if (req_offset + req_len > blob_size) {
+                LOGERROR("Invalid offset length request in get blob {} offset {} len {} size {}", blob_id, req_offset,
+                         req_len, blob_size);
+                iomanager.iobuf_free(iov_base);
+                return folly::makeUnexpected(BlobError::INVALID_ARG);
+            }
+
+            // Copy the blob bytes from the offset. If request len is 0, take the
+            // whole blob size else copy only the request length.
+            blob_bytes += req_offset;
+            auto res_len = req_len == 0 ? blob_size : req_len;
+            auto body = sisl::io_blob_safe(res_len);
+            std::memcpy(body.bytes, blob_bytes, res_len);
```

Comment:
will there be a user case that req_len == 0 and req_offset != 0?
i know this is a wrong case , but it might happens in some case . so i think we should check this before `memcpy` , otherwise, segment fault will happen

Reply:
just make L197 to 
```
auto res_len = req_len == 0 ? blob_size-req_offset : req_len;
```

Reply:
Ack